frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button } from '#libs/components/components.js';
+import { IconColor } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useHandleClickOutside, useRef } from '#libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: React.ReactNode;
+  isDisplayed: boolean;
+  title: string;
+  onClose: () => void;
+};
+
+const Modal: React.FC<Properties> = ({
+  children,
+  isDisplayed,
+  title,
+  onClose,
+}) => {
+  const reference = useRef<HTMLDivElement>(null);
+
+  useHandleClickOutside({ reference, onClose });
+
+  return (
+    <dialog
+      open={isDisplayed}
+      className={getValidClassNames(isDisplayed && styles['overlay'])}
+    >
+      <div className={styles['modal']} ref={reference}>
```

Comment:
There're actually will be 2 refs, one on the overlay and second as it's now, so we won't use `isDispalyed` state at all